step-automation-packages/step-automation-packages-junit/src/main/java/step/engine/plugins/AutomationPackageKeywordsPlugin.java

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ * Copyright (C) 2020, exense GmbH
+ *
+ * This file is part of STEP
+ *
+ * STEP is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * STEP is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with STEP.  If not, see <http://www.gnu.org/licenses/>.
+ ******************************************************************************/
+package step.engine.plugins;
+
+import ch.exense.commons.app.Configuration;
+import step.automation.packages.AutomationPackageArchive;
+import step.automation.packages.AutomationPackageKeywordsAttributesApplier;
+import step.automation.packages.AutomationPackageReader;
+import step.automation.packages.AutomationPackageReadingException;
+import step.automation.packages.model.AutomationPackage;
+import step.automation.packages.model.AutomationPackageKeyword;
+import step.core.execution.AbstractExecutionEngineContext;
+import step.core.execution.ExecutionEngineContext;
+import step.core.execution.OperationMode;
+import step.core.plugins.Plugin;
+import step.functions.Function;
+import step.functions.accessor.FunctionAccessor;
+import step.resources.ResourceManager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Plugin(dependencies = {FunctionPlugin.class, AutomationPackageJsonSchemaPlugin.class})
+public class AutomationPackageKeywordsPlugin extends AbstractExecutionEnginePlugin {
+
+    private FunctionAccessor functionAccessor;
+    private ResourceManager resourceManager;
+
+    @Override
+    public void initializeExecutionEngineContext(AbstractExecutionEngineContext parentContext, ExecutionEngineContext context) {
+        if (context.getOperationMode() == OperationMode.LOCAL) {
+            functionAccessor = context.require(FunctionAccessor.class);
+            resourceManager = context.getResourceManager();
+
+            List<Function> localFunctions = getFunctionsFromAutomationPackage(context.getConfiguration());
+            functionAccessor.save(localFunctions);
+        }
+    }
+
+    public List<Function> getFunctionsFromAutomationPackage(Configuration configuration) {
+        ArrayList<Function> res = new ArrayList<>();
+
+        // the automation package should be found in current classloader
+        AutomationPackageArchive automationPackageArchive = new AutomationPackageArchive(this.getClass().getClassLoader());
```

Comment:
@jeromecomte 
In fact I've completely forgotten about referencing plans from automation package in `@Plans` annotation.

For now we only can reference separate `.plan` or `.yml` plans (see AutomationPackageKeywordsPluginTest), which are not listed in automation package descriptor. But in these plans we can use keywords from `automation.yml`.

I think, if we want to call the plans defined in automation.yml, we have to extend the @Plans annotation and allow to use plan names from automation.yml there. In this case we can prepare both keywords and plans in `AutomationPackageKeywordsPlugin` and in `step.junit.runner.Step` we can reuse there plans instead of calling the `StepClassParser`.

Reply:
@iegorov777 Adding this point to the agenda of today's meeting

Reply:
As discussed during our call we'll decide how to handle the @Plans annotation after merging this PR. Resolving